lib/engine/game/g_18_mag/game.rb

```diff
@@ -616,31 +619,47 @@ def check_distance(route, visits)
           raise GameError, 'Must visit minimum of two non-mine stops' if visits.sum(&:visit_cost) < 2
         end
 
+        def compute_stops(route, train = nil)
+          train ||= route.train
+          distance = if gc_train?(route) && !other_gc_train?(route)
+                       gc_train_distance(route.train.distance)
+                     else
+                       route.train.distance
+                     end
+          train_dup = train.dup
+          train_dup.distance = distance
+
+          super(route, train_dup)
+        end
+
         def gc_train_distance(route_distance)
           if route_distance.is_a?(Numeric)
             town_distance_value = route_distance
+            city_distance_value = route_distance
           else
             # route has a 1 town value from the supporter
             town_distance = route_distance.find { |n| n['nodes'] == ['town'] }
-            town_distance_value = town_distance['pay'] + route_distance
+            city_distance_value = route_distance.find { |n| n['nodes'].include?('city') }['pay']
+            town_distance_value = town_distance['pay'] + city_distance_value
           end
           [
-              {
-                nodes: %w[city offboard town],
-                pay: route_distance,
-                visit: route_distance,
-              },
-              {
-                nodes: %w[town],
-                pay: town_distance_value,
-                visit: town_distance_value,
-              },
+            {
+              nodes: %w[town],
+              pay: town_distance_value,
+              visit: town_distance_value,
+            },
+            {
+              nodes: %w[city offboard town],
+              pay: city_distance_value,
+              visit: city_distance_value,
+            },
+
```

Comment:
Does changing the order between town and city,offboard,town matter?

Reply:
so as far as i know the rules of 18mag allow for a 2+2 train (after upgrade from the major) to run 1 city and 3 towns.
If that's the case than the correct order should be town then city/town/offboard.